All_Relationships/2.data-labeler.py

```diff
@@ -0,0 +1,221 @@
+
+# coding: utf-8
+
+# # MUST RUN AT THE START OF EVERYTHING
+
+# Load all the imports and set up the database for database operations. Plus, set up the particular candidate type this notebook is going to work with. 
+
+# In[2]:
+
+
+get_ipython().magic(u'load_ext autoreload')
+get_ipython().magic(u'autoreload 2')
+get_ipython().magic(u'matplotlib inline')
+
+from collections import defaultdict
+import re
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import tqdm
+
+
+# In[ ]:
+
+
+#Set up the environment
+username = "danich1"
+password = "snorkel"
+dbname = "pubmeddb"
+
+#Path subject to change for different os
+database_str = "postgresql+psycopg2://{}:{}@/{}?host=/var/run/postgresql".format(username, password, dbname)
+os.environ['SNORKELDB'] = database_str
+
+from snorkel import SnorkelSession
+session = SnorkelSession()
+
+
+# In[ ]:
+
+
+from snorkel.annotations import FeatureAnnotator, LabelAnnotator
+from snorkel.features import get_span_feats
+from snorkel.models import candidate_subclass
+from snorkel.models import Candidate
+from snorkel.viewer import SentenceNgramViewer
+
+
+# In[ ]:
+
+
+edge_type = "dg"
+debug = False
+
+
+# In[ ]:
+
+
+if edge_type == "dg":
+    DiseaseGene = candidate_subclass('DiseaseGene', ['Disease', 'Gene'])
+    edge = "disease_gene"
+elif edge_type == "gg":
+    GeneGene = candidate_subclass('GeneGene', ['Gene1', 'Gene2'])
```

Comment:
What is the motivation here to do `['Gene1', 'Gene2']` rather than `['Gene', 'Gene']`?

Reply:
So for the candidate subclass it needs to take two column names and they need to be unique; therefore, I appended the 1 and 2. I got an error when I used `['Gene', 'Gene']`.